### PR TITLE
Expectations

### PR DESCRIPTION
**Pull request by bogdanmarculescu**
Part of the wrapping up.

Two types of expectations are generated:
1. expectations about the content (activated by expectationsMasterSwitch) which should duplicate basic assertions.
2. expectations about the structure (activated by responseStructureOracle) which make statements about the expected structure of returned objects (where the swagger has such objects to begin with).

Not sure why the CI assessments are not run at the moment.

### CODE REVIEW

**Comment by arcuri82** [comment]
hi,
 all CI checks are failing. please look into them

File: core/src/main/kotlin/org/evomaster/core/output/service/PartialOracles.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.google.gson.Gson
+import org.evomaster.core.output.Lines
+import org.evomaster.core.problem.rest.RestCallAction
+import org.evomaster.core.problem.rest.RestCallResult
+import org.evomaster.core.search.gene.ObjectGene
+import org.evomaster.core.search.gene.OptionalGene
+
+
+class PartialOracles {
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/output/service/PartialOracles.kt]: provide some comments explaining what are the the partial oracles

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/contentMatchers/StringMatcher.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package org.evomaster.client.java.controller.contentMatchers;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+public class StringMatcher extends TypeSafeMatcher<String> {
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/contentMatchers/StringMatcher.java]: can  u add some test cases for this new class?

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/contentMatchers/SubStringMatcher.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package org.evomaster.client.java.controller.contentMatchers;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+public class SubStringMatcher  extends TypeSafeMatcher<String> {
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/contentMatchers/SubStringMatcher.java]: can  u add some test cases for this new class?

File: core/src/main/kotlin/org/evomaster/core/EMConfig.kt [hunk 4]
```diff
@@ -703,7 +703,7 @@ class EMConfig {
 
     @Experimental
     @Cfg("Enable Expectation Generation. If enabled, expectations will be generated. " +
-            "A variable called activeExpectations is added to each test case, with a default value of false. If set to true, an expectation that fails will cause the test case containing it to fail.")
+            "A variable called expectationsMasterSwitch is added to the test suite, with a default value of false. If set to true, an expectation that fails will cause the test case containing it to fail.")
     var expectationsActive = false
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/EMConfig.kt]: are the e2e tests passing when this is set to `true`?

File: core/src/main/kotlin/org/evomaster/core/output/service/PartialOracles.kt (hunk 1 above)
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/output/service/PartialOracles.kt]: Will do. I'm expecting this to change quite a bit, but I will do my best to have the explanation there and to keep it as up to date as possible.

File: core/src/main/kotlin/org/evomaster/core/EMConfig.kt (hunk 4 above)
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/EMConfig.kt]: Generated tests pass, yes.
EM's e2e tests fail (at least the TestCaseWriterTest does). I'll adjust that and see if anything else causes a problem.

**Comment by bogdanmarculescu** [comment]
Hej. I think I've covered the required changes. The CI tests seem to pass. (Note that for the tests regarding TestSuiteWriter and TestCaseWriter, I forced the expectationsEnabled option to false).

Do you intend to have the expectations enabled by default?

**Comment by arcuri82** [comment]
hi. thx. we can wait a bit before before putting them on by default (want to do some manual testing first, later this week)

**Pull request merged**